src/cmd/doctor.js

```diff
@@ -67,14 +68,38 @@ module.exports = class DoctorCommand {
 		}
 	}
 
+	async getDfuDevices() {
+		const devices = await usbUtils.getUsbDevices({ dfuMode: true });
+		const dfuDevices = [];
+		for (const device of devices) {
+			if (device.isInDfuMode) {
+				const d = this._getDfuId(device);
+				dfuDevices.push({
+					type: deviceSpecs[d].productName,
+					dfuId: d,
+					specs: deviceSpecs[d]
+				});
+			}
+		}
+		return dfuDevices;
+	}
+
+	_getDfuId(device) {
+		// TODO: Remove the usage of dfuId and use device-constants
+		const vendorId = device._info.vendorId;
+		const productId = device._info.productId;
+		return vendorId.toString(16).padStart(4, '0') + ':' + productId.toString(16).padStart(4, '0');
+	}
+
 	_findDevice(){
+		// TODO: Replace the serial commands to use usb-utils
+
 		// Try to find a "normal" mode device through the serial port
 		return this.command('serial')
 			.findDevices()
 			.then(devices => {
 				if (devices.length === 0){
-					// Try to find a "DFU" mode device through dfu-util
-					return dfu.listDFUDevices();
+					return this.getDfuDevices();
```

Comment:
I replaced the `dfu.listDFUDevices` to use usbUtils in such a way that this line exactly returns the object that was previously returned by dfu.listDFUDevices.

Pasting the return values from 
--> getDfuDevices of this PR
[getDfuDevicesRes Full output](https://github.com/particle-iot/particle-cli/files/12738191/getDfuDevicesRes.txt)

```
[
  {
    "type": "Photon 2 / P2",
    "dfuId": "2b04:d020",
    "specs": {
      .....
      }
    }
  },
  {
    "type": "Argon",
    "dfuId": "2b04:d00c",
    "specs": {
      ......
      }
    }
  }
]
```

---> From dfu.listDFUDevices [listDFUDevices full output](https://github.com/particle-iot/particle-cli/files/12738310/listDFUDevicesResp.txt)

```
[
  {
    "type": "Argon",
    "dfuId": "2b04:d00c",
    "specs": {
      .....
    }
  },
  {
    "type": "Photon 2 / P2",
    "dfuId": "2b04:d020",
    "specs": {
     ....
    }
  }
]
```